app/login/page.tsx

```diff
@@ -0,0 +1,115 @@
+"use client"
+import { useRouter } from "next/navigation"
+import { Button } from "@/components/uikit/button"
+import { Input } from "@/components/uikit/input"
+import { useFormik } from "formik"
+import toast from "react-hot-toast"
+import * as Yup from "yup"
+import { login, signUp } from "@/domains/auth/services"
+
+export default function Login() {
+  const router = useRouter()
+
+  const formik = useFormik({
+    initialValues: {
+      name: "",
+      email: "",
+      password: "",
+    },
+
+    validationSchema: Yup.object({
+      name: Yup.string()
+        .max(50, "Must be 50 characters or less")
+        .required("Обов'язкове поле"),
+      email: Yup.string().required("Обов'язкове поле"),
+      password: Yup.string().min(7).required("Обов'язкове поле"),
+    }),
+    onSubmit: (values) => {
+      signUp(values)
+      toast.success("Особистий кабінет створено")
+      router.push(`/cards`)
+    },
+  })
+
+  const formikSignIn = useFormik({
+    initialValues: {
+      email: "",
+      password: "",
+    },
+
+    validationSchema: Yup.object({
+      email: Yup.string().required("Обов'язкове поле"),
+      password: Yup.string().min(7).required("Обов'язкове поле"),
+    }),
+    onSubmit: (values) => {
+      login(values.email, values.password)
+      router.push(`/cards`)
+    },
+  })
+  const { getFieldProps } = formik
+  return (
+    <div className="mt-5 text-center">
+      <form className="flex-col  " onSubmit={formik.handleSubmit}>
+        <Input {...getFieldProps("name")} label="Ім'я" type="text" />
+        {formik.touched.name && formik.errors.name ? (
+          <div className="text-red-500 text-xs">{formik.errors.name}</div>
+        ) : null}
```

Comment:
Let's try to pass `error` to `Input`.

`Input.tsx`
```tsx
import { cn } from "@/utils"
import React from "react"

type Props = React.InputHTMLAttributes<HTMLInputElement> & {
  label?: React.ReactNode
  error?: React.ReactNode
}

const _Input = (props: Props, ref: React.ForwardedRef<HTMLInputElement>) => {
  const { label, className, error, ...restProps } = props

  return (
    <div className={cn("flex flex-col items-center", className)}>
      <label>{label}</label>
      <input
        {...restProps}
        ref={ref}
        className="border-2 border-sky-400 h-10 w-2/3"
      />
      {error ? <p className="text-red-500 text-xs">{error}</p> : null}
    </div>
  )
}

export const Input = React.forwardRef(_Input)
```

Also, please, pay attention to the forwarded ref. [Read more](https://react.dev/reference/react/forwardRef).